packages/big-design/src/components/Dropdown/types.ts

```diff
@@ -6,7 +6,7 @@ export type DropdownOption<T> = DropdownItem<T> | DropdownLinkItem<T>;
 
 export interface DropdownProps<T> extends Omit<React.HTMLAttributes<HTMLUListElement>, 'children'> {
   maxHeight?: number;
-  options: Array<DropdownOption<T>>;
+  options: Array<DropdownOption<T> | DropdownOptionGroup<T>>;
```

Comment:
Should this be `Array<DropdownOptionGroup<T>>`? 🤔

Reply:
Both the `DropdownOption<T>` and the `DropdownOptionGroup<T>` are inside the `Array<>` distinction. I could also separate them to read `Array<DropdownOption<T>> | Array<DropdownOptionGroup<T>>` - is that what you mean?

Reply:
Ahh I missed that, however, I think your second proposal makes more sense.

If we do `Array<DropdownOption<T> | DropdownOptionGroup<T>>` that means you could mix both DropdownOptionGroup and DropdownOptions in the same array. Since your code is checking if all are options or all are groups (which makes sense) then the type is a bit off.

If we do `Array<DropdownOption<T>> | Array<DropdownOptionGroup<T>>` we would ensure that you can only use one kind.

Reply:
Oooh good call, thanks for pointing that out 🙌